internal/processtags/processtags.go

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package processtags
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/traceutil"
+
+	"github.com/DataDog/dd-trace-go/v2/internal"
+	"github.com/DataDog/dd-trace-go/v2/internal/log"
+)
+
+const (
+	tagEntrypointName    = "entrypoint.name"
```

Comment:
there is also another tag `entrypoint.type` that can eventually be set here or in another PR.  For go there is a single value `executable`